File: TicketTriageUI/services/api_client.py
import requests

BASE_URL = "https://xesajug973.execute-api.us-east-1.amazonaws.com/Prod"




#pOST TICKET(Creating ticket)
def create_ticket(payload):

    try:

        response = requests.post(
            f"{BASE_URL}/tickets",
            json=payload
        )

        response.raise_for_status()

        return response.json()

    except Exception as e:

        raise Exception(
            f"Failed to create ticket: {e}"
        )
    


#GET TICKETS(Fetching tickets)
def get_tickets():
    
    response = requests.get(
        f"{BASE_URL}/tickets"
    )

    response.raise_for_status()

    return response.json()


# GET TICKET BY ID(Each ticket)
def get_ticket_by_id(ticket_id):

    response = requests.get(
        f"{BASE_URL}/tickets/{ticket_id}"
    )

    response.raise_for_status()

    return response.json()

# 
def get_approvals():
    
    response = requests.get(
        f"{BASE_URL}/approvals"
    )

    response.raise_for_status()

    return response.json()

#
def approve_request(approval_id):
    
    response = requests.get(
        f"{BASE_URL}/approve",
        params={
            "approvalId": approval_id,
            "action": "approve"
        }
    )
    response.raise_for_status()

    return response.json()


def reject_request(approval_id):
    
    response = requests.get(
        f"{BASE_URL}/reject",
        params={
            "approvalId": approval_id,
            "action": "reject"
        }
    )

    response.raise_for_status()

    return response.json()


# GET ORDERS
def get_orders():

    response = requests.get(
        f"{BASE_URL}/orders"
    )

    response.raise_for_status()

    return response.json()

File: TicketTriageUI/src/api/apiClient.js
import axios from "axios";

const BASE_URL = import.meta.env.VITE_API_BASE_URL;

const http = axios.create({
  baseURL: BASE_URL,
});

// POST TICKET (Creating ticket)
export async function createTicket(payload) {
  try {
    const response = await http.post("/tickets", payload);
    return response.data;
  } catch (e) {
    throw new Error(`Failed to create ticket: ${describeError(e)}`);
  }
}

// GET TICKETS (Fetching tickets)
export async function getTickets() {
  const response = await http.get("/tickets");
  return response.data;
}

// GET TICKET BY ID (Each ticket)
export async function getTicketById(ticketId) {
  const response = await http.get(`/tickets/${ticketId}`);
  return response.data;
}

export async function getApprovals() {
  const response = await http.get("/approvals");
  return response.data;
}

export async function approveRequest(approvalId) {
  const response = await http.get("/approve", {
    params: { approvalId, action: "approve" },
  });
  return response.data;
}

export async function rejectRequest(approvalId) {
  const response = await http.get("/reject", {
    params: { approvalId, action: "reject" },
  });
  return response.data;
}

// GET ORDERS
export async function getOrders() {
  const response = await http.get("/orders");
  return response.data;
}

function describeError(e) {
  return e.response?.data?.message || e.message || String(e);
}
